web/server/codechecker_server/auth/cc_ldap.py

```diff
@@ -139,22 +139,41 @@ def ldap_error_handler():
 def get_user_dn(con,
                 account_base_dn,
                 account_pattern,
-                scope=ldap.SCOPE_SUBTREE):
+                scope=ldap.SCOPE_SUBTREE,
+                user_dn_postfix_preference=None):
     """
     Search for the user dn based on the account pattern.
     Return the full user dn None if search failed.
+
+    user_dn_postfix_preference
+    User DN postfix preference value can be used to select out one prefered
+    user DN if multiple DN entries are found buy the LDAP search.
+    The configured value will be matched and the first matching will be used.
+    If only one DN was found this postfix mathcing will not be used.
     """
 
     with ldap_error_handler():
         # Attribute values MAY contain any type of data. Before you use a
         # value, call 'bytes_to_str' helper function to convert it to text.
         user_data = con.search_s(account_base_dn, scope, account_pattern)
-
+        user_dns = []
         if user_data:
             # User found use the user DN from the first result.
-            user_dn = bytes_to_str(user_data[0][0])
-            LOG.debug("Found user: %s", user_dn)
-            return user_dn
+            if len(user_data) > 1:
+                for user_info in user_data:
+                    user_dns.append(bytes_to_str(user_info[0]))
+            else:
+                user_dns.append(bytes_to_str(user_data[0][0]))
```

Comment:
I think we can simplify this part the following way:
```suggestion
            for user_info in user_data:
                user_dns.append(bytes_to_str(user_info[0]))
```